db/migrate/20240423054323_create_flags.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class CreateFlags < ActiveRecord::Migration[7.0]
+  def change
+    create_table :flags do |t|
+      t.string :name, unique: true
+      t.integer :position, null: false
+      t.boolean :enabled, default: true, null: false
+      t.boolean :topic_type, default: false, null: false
+      t.boolean :notify_type, default: false, null: false
+      t.boolean :auto_action_type, default: false, null: false
+      t.boolean :custom_type, default: false, null: false
```

Comment:
@martin-brennan what do you think about adding `chat_type`? Even if right now, all custom flags will be working with chat as well, later we will be easily able to modify UI to be more granular.

Reply:
And what makes a flag be `custom_type` true? It seems very arbitrary right now.

Reply:
This can mostly be disregarded now, see latest comment. We still do need to think about how notify/auto action stuff will work down the road though, which may require migrations.